packages/analytics-core/src/plugins/destination.ts

```diff
@@ -78,14 +74,19 @@ export class Destination implements DestinationPlugin {
         event,
         attempts: 0,
         callback: (result: Result) => resolve(result),
-        timeout: 0,
       };
       void this.addToQueue(context);
     });
   }
 
-  getTryableList(list: Context[]) {
-    return list.filter((context) => {
+  addToQueue(...list: Context[]) {
+    this.queue = this.queue.concat(list);
+    this.sendEventIfReady();
```

Comment:
Should we save before we send? This is to make sure an error in `sendEventIfReady` does not drop data

Reply:
Moved the saveEvents before sendEventIfReady.